src/d3a/models/area/__init__.py

```diff
@@ -403,29 +402,16 @@ def tick(self):
             if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
                 self.dispatcher.publish_market_clearing()
             else:
-                self._match_bids_offers()
+                self._update_myco_matcher()
 
         self.events.update_events(self.now)
 
-    def _match_bids_offers(self) -> None:
-        """Match bids and offers for all markets."""
-        if is_external_matching_enabled():
-            # Update the open offer bids cache that the myco client will request
-            bid_offer_matcher.match_algorithm.update_area_uuid_markets_mapping(
-                {self.uuid: self.all_markets})
-            return
-        # If the external matching is not enabled, get and match bid offer pairs
-        for market in self.all_markets:
-            while True:
-                bids, offers = market.open_bids_and_offers
-                data = {
-                    market.id: {"bids": [bid.serializable_dict() for bid in bids.values()],
-                                "offers": [offer.serializable_dict() for offer in offers.values()],
-                                "current_time": self.now}}
-                bid_offer_pairs = bid_offer_matcher.get_matches_recommendations(data)
-                if not bid_offer_pairs:
-                    break
-                market.match_recommendations(bid_offer_pairs)
+    def _update_myco_matcher(self) -> None:
+        """Update the markets cache that the myco matcher will request"""
+        bid_offer_matcher.matcher.update_area_uuid_markets_mapping(
+            area_uuid_markets_mapping={self.uuid: {"markets": self.all_markets, "current_time": self.now}})
+        if not is_external_matching_enabled():
+            bid_offer_matcher.matcher.match_recommendations()
```

Comment:
This implementation violates the single responsibilty principle. Please refactor into separate functions (or just move 
``` python
if not is_external_matching_enabled():
            bid_offer_matcher.matcher.match_recommendations()
```
out of here. Thanks!